source/agora/node/BlockStorage.d

```diff
@@ -742,9 +742,9 @@ public class BlockStorage : IBlockStorage
             File idx_file = File(this.index_path, "a+b");
             idx_file.seek(0, SEEK_END);
 
-            serializePart(height, (scope v) => idx_file.rawWrite(v));
+            serializePart(height, (scope v) => idx_file.rawWrite(v), false);
```

Comment:
Why is this one not compact? Is there a special reason?

Reply:
According to Blockstorge Definition done,
Blockstore index item should be a fixed length.
However, it is not possible to perform index role if compact size is applied.
This part caused an error in the test.
A separate special case has been defined.

Reply:
Ok.

Could you make it an enum?

```D
enum CompactMode
{
    No,
    Yes
}
```

And then call with `serializePart(height, (scope v) => idx_file.rawWrite(v), CompactMode.No);`

And:

```D
public void deserializePart (ref ulong record, scope DeserializeDg dg, CompactMode compact = CompactMode.Yes)
```

> According to Blockstorge Definition done, Blockstore index item should be a fixed length.

Well ideally yes, but in the current implementation of BlockStorage the height/position and hash/position are read one at a time on start-up and stored into an RBTree. So from what I can tell, there is currently no benefit for having the index be fixed-length. 

But you can keep it fixed-length for now.

Reply:
Applied the suggestion.